DllImportGenerator/DllImportGenerator/StubCodeGenerator.cs

```diff
@@ -138,31 +144,44 @@ public override (string managed, string native) GetIdentifiers(TypePositionInfo
             // Stub return is not the same as invoke return
             if (!stubReturnsVoid && !retMarshaller.TypeInfo.IsManagedReturnPosition)
             {
+                // Stub return should be the last parameter for the invoke
                 Debug.Assert(paramMarshallers.Any() && paramMarshallers.Last().TypeInfo.IsManagedReturnPosition);
 
-                // Declare variable for stub return value
-                TypePositionInfo info = paramMarshallers.Last().TypeInfo;
-                statements.Add(LocalDeclarationStatement(
-                    VariableDeclaration(
-                        info.ManagedType.AsTypeSyntax(),
-                        SingletonSeparatedList(
-                            VariableDeclarator(this.GetIdentifiers(info).managed)))));
+                (TypePositionInfo stubRetTypeInfo, IMarshallingGenerator stubRetGenerator) = paramMarshallers.Last();
+                if (stubRetGenerator.UsesNativeIdentifier(stubRetTypeInfo, this))
+                {
+                    // Update the native identifier for the return value
+                    ReturnNativeIdentifier = $"{ReturnIdentifier}{GeneratedNativeIdentifierSuffix}";
+                }
+
+                // Declare variables for stub return value
+                AddVariableDeclations(setupStatements, stubRetTypeInfo, stubRetGenerator);
             }
 
             if (!invokeReturnsVoid)
             {
-                // Declare variable for invoke return value
-                statements.Add(LocalDeclarationStatement(
-                    VariableDeclaration(
-                        retMarshaller.TypeInfo.ManagedType.AsTypeSyntax(),
-                        SingletonSeparatedList(
-                            VariableDeclarator(this.GetIdentifiers(retMarshaller.TypeInfo).managed)))));
+                // Declare variables for invoke return value
+                AddVariableDeclations(setupStatements, retMarshaller.TypeInfo, retMarshaller.Generator);
+            }
+
+            var tryStatements = new List<StatementSyntax>();
+            var finallyStatements = new List<StatementSyntax>();
+            List<StatementSyntax> GetStatements(Stage stage)
```

Comment:
I would prefer local functions at the end of the function typically. Is there a strong reason for in the middle of the enclosing function?